lib/simple-connect-policy.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+
+const EventEmitter = require('events').EventEmitter;
+
+// Simple generic connection provider. Used for user-side connection.
+// You are free to implement whatever suits your needs instead.
+// Sends handshake with login/password if provided otherwise sends
+// anonymous handshake.
+//
+module.exports = class SimpleConnectPolicy extends EventEmitter {
```

Comment:
Does it really even need to be a class, not to mention an EventEmitter (I don't see any EventEmitter functionality used here)? Requiring this and the auth policy to be just functions seems both simpler and more flexible to me (one can always pass a bound method if they need it to be stateful).